src/Compatibility/ApiCompat/Microsoft.DotNet.ApiCompat.Shared/ApiCompatServiceProvider.cs

```diff
@@ -27,21 +27,12 @@ public ApiCompatServiceProvider(Func<ISuppressionEngine, ISuppressibleLog> logFa
             _compatibilityLogger = new Lazy<ISuppressibleLog>(() => logFactory(SuppressionEngine));
             _apiCompatRunner = new Lazy<IApiCompatRunner>(() =>
             {
-                AccessibilitySymbolFilter accessibilitySymbolFilter = new(respectInternals);
```

Comment:
We intentionally shared this filter before so that we only need to instantiate one. Is the filter still shared with your changes?

Reply:
It is not, this filter instance is now internally created in `SymbolFilterFactory`.

Reply:
There was no caching before, and a new accessibility symbol filter instance was being created in the ApiComparerSettings constructor anyway.

Reply:
> and a new accessibility symbol filter instance was being created in the ApiComparerSettings constructor anyway.

I don't think that is true. One only gets created if null gets passed in. We passed the shared filter in before.

Reply:
I'll add an extra optional argument to the SymbolFilterFactory methods so they take an accessbilitySymbolFilter if we want to avoid multiple instances.